File: solutions/js/day1-send-xrp.js
// ============================================================
// Day 1 Solution: Send XRP
// Learn: How to construct and submit a Payment transaction
// on the XRP Ledger testnet.
// ============================================================

const xrpl = require('xrpl');
const fs = require('fs');

const TESTNET_URL = 'wss://s.altnet.rippletest.net:51233';

async function main() {
  const { wallets } = JSON.parse(fs.readFileSync('../../wallets.json', 'utf-8'));
  const senderWallet = xrpl.Wallet.fromSeed(wallets[0].seed);
  const receiverWallet = xrpl.Wallet.fromSeed(wallets[1].seed);

  const client = new xrpl.Client(TESTNET_URL);
  await client.connect();
  console.log('Connected to XRPL Testnet');

  const balanceBefore = await client.getXrpBalance(receiverWallet.address);
  console.log(`Receiver balance before: ${balanceBefore} XRP`);

  const paymentTx = {
    TransactionType: 'Payment',
    Account: senderWallet.address,
    Amount: xrpl.xrpToDrops('10'),
    Destination: receiverWallet.address
  };

  const result = await client.submitAndWait(paymentTx, { wallet: senderWallet });
  console.log(result);

  const balanceAfter = await client.getXrpBalance(receiverWallet.address);
  console.log(`Receiver balance after: ${balanceAfter} XRP`);

  console.log(`\nExplorer: https://testnet.xrpl.org/accounts/${senderWallet.address}`);
  await client.disconnect();
}

main().catch(console.error);

File: solutions/python/day1_send_xrp.py
"""
Day 1 Solution: Send XRP
=========================
Send 10 XRP from wallet[0] to wallet[1] on XRPL Testnet.
"""

import json
from xrpl.clients import WebsocketClient
from xrpl.wallet import Wallet
from xrpl.models.transactions import Payment
from xrpl.transaction import submit_and_wait
from xrpl.utils import xrp_to_drops

TESTNET_URL = 'wss://s.altnet.rippletest.net:51233'

# Load wallets
with open('../../wallets.json') as f:
    data = json.load(f)
    wallets = data['wallets']

sender = Wallet.from_seed(wallets[0]['seed'])
receiver = Wallet.from_seed(wallets[1]['seed'])

print(f"Sender:   {sender.address}")
print(f"Receiver: {receiver.address}")

with WebsocketClient(TESTNET_URL) as client:

    # Create Payment transaction
    payment_tx = Payment(
        account=sender.address,
        amount=xrp_to_drops(10),
        destination=receiver.address
    )

    # Submit and print result
    result = submit_and_wait(payment_tx, client, sender)
    print(result)

# Explorer: https://testnet.xrpl.org/accounts/{address}
print(f"\nCheck sender:   https://testnet.xrpl.org/accounts/{sender.address}")
print(f"Check receiver: https://testnet.xrpl.org/accounts/{receiver.address}")
